### PR TITLE
GEODE-9804: Fixes incorrect fetching of initial entries.

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**
* Makes InterestResultPolicy a proper enum.
* Adds integration benchmark for register interest.
* Cleanup methods unused arguments.

### CODE REVIEW

File: cppcache/src/ThinClientRegion.cpp [hunk 1]
```diff
@@ -2208,45 +2149,40 @@ GfErrType ThinClientRegion::registerKeysNoThrow(
...
-    if (needToCreateRC) {
-      resultCollector = (new ChunkedInterestResponse(request, nullptr, *reply));
-      reply->setChunkedResultHandler(resultCollector);
+  std::unique_ptr<TcrChunkedResult> resultCollector;
+  TcrMessageReply replyLocal(true, m_tcrdm.get());
+  if (!reply) {
+    reply = &replyLocal;
+    if (interestPolicy == InterestResultPolicy::KEYS_VALUES) {
+      auto values = std::make_shared<HashMapOfCacheable>();
+      auto exceptions = std::make_shared<HashMapOfException>();
```
Comment (pdxcodemonkey) [inline cppcache/src/ThinClientRegion.cpp:2174]: Why is this map here?  Can we push it down into the `ChunkedGetAllResponse` ctor?

File: cppcache/src/ThinClientRegion.cpp [hunk 2]
```diff
@@ -2384,64 +2317,51 @@ GfErrType ThinClientRegion::registerRegexNoThrow(
...
+  } else {
     reply = &replyLocal;
-    if (interestPolicy.ordinal == InterestResultPolicy::KEYS_VALUES.ordinal) {
+    if (!resultKeys) {
+      resultKeys =
+          std::make_shared<std::vector<std::shared_ptr<CacheableKey>>>();
+    }
+    if (interestPolicy == InterestResultPolicy::KEYS_VALUES) {
+      auto values = std::make_shared<HashMapOfCacheable>();
+      auto exceptions = std::make_shared<HashMapOfException>();
```
Comment (pdxcodemonkey) [inline cppcache/src/ThinClientRegion.cpp:2348]: Same here - exceptions map looks like it's just a transient parameter to the `ChunkedGetAllResponse` ctor

**Review by pdxcodemonkey — CHANGES_REQUESTED** [review]
Just a couple of small things and a few questions to answer.  Thanks, this is a big improvement!

**Comment by pivotal-jbarrett** [comment]
Closing in favor of https://github.com/apache/geode-native/pull/894

**Pull request closed without merging**